src/Cli/dotnet/SlnProjectExtensions.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.Sln.Internal;
+
+namespace Microsoft.DotNet.Tools.Common
+{
+    internal static class SlnProjectExtensions
+    {
+        public static string GetFullSolutionFolderPath(this SlnProject slnProject)
+        {
+            var slnFile = slnProject.ParentFile;
+            var nestedProjects = slnFile.Sections
+                .GetOrCreateSection("NestedProjects", SlnSectionType.PreProcess)
+                .Properties;
```

Comment:
Do we know that all of the things in `_properties` under a solution folder will be projects? To me it looks like it, but I don't know for sure.

Reply:
I confirmed with RichardS that everything in the nested project section is a project (either a solution folder or project but in the solution file itself, they are all considered projects).